pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -685,30 +708,35 @@ func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger
 		// be empty. If it is not, we may have a problem.
 		if len(pInfo.UnschedulablePlugins) != 0 {
 			logger.Error(nil, "In flight Pod isn't found in the scheduling queue. If you see this error log, it's likely a bug in the scheduler.", "pod", klog.KObj(pInfo.Pod))
-			return queueAfterBackoff
+			return nil, queueAfterBackoff
 		}
 		if p.inFlightEvents.Len() > len(p.inFlightPods) {
-			return queueAfterBackoff
+			return nil, queueAfterBackoff
 		}
-		return queueSkip
+		return nil, queueSkip
```

Comment:
I know it's an original issue unrelated to the PR's change, but I do not understand why we have to get all these `if`s.
They're added in https://github.com/kubernetes/kubernetes/pull/120413/.
@pohly do you remember, by any chance..? a year ago though..

From my eyes, we must have the pod in `inFlightPods` always; in other words, if not, we should always log an error like L710 and put the pod back to activeQ/backoffQ.

Reply:
https://github.com/kubernetes/kubernetes/pull/120413/files#r1315165907

Worth checking if it's still the case.

Reply:
I confirmed that we can just remove those `if`s: https://github.com/kubernetes/kubernetes/pull/125930